packages/beacon-node/src/chain/blocks/importBlock.ts

```diff
@@ -95,15 +95,20 @@ export async function importBlock(
   this.logger.verbose("Added block to forkchoice and state cache", {slot: blockSlot, root: blockRootHex});
 
   // We want to import block asap so call all event handler in the next event loop
-  setTimeout(() => {
+  setTimeout(async () => {
     this.emitter.emit(routes.events.EventType.block, {
       block: blockRootHex,
       slot: blockSlot,
       executionOptimistic: blockSummary != null && isOptimisticBlock(blockSummary),
     });
 
-    if (blockInput.type === BlockInputType.postDeneb) {
-      for (const blobSidecar of blockInput.blobs) {
+    if (blockInput.type === BlockInputType.postDeneb || blockInput.type === BlockInputType.blobsPromise) {
```

Comment:
we should have full blobs here to reach `importBlock` phase
maybe throw error if `BlockInputType.blobsPromise`?

Reply:
just review again, a `availabilityPromise` block input could reach this `importBlock` because we eagerly process block when blobs are not fully received

ideally at `verifyBlocksInEpoch` we could process `availabilityPromise` BlockInput but after that function I'd like to receive a full blobs (`postDeneb`) BlockInput to be sent to `importBlock`, is it possible @g11tech ?
curious about your thought on this @wemeetagain 

it's not related to this PR, just want to improve a little bit